site/app/controllers/student/TeamController.php

```diff
@@ -496,6 +508,22 @@ public function setTeamName($gradeable_id): RedirectResponse {
         if ($_POST['team_name'] == '') {
             $_POST['team_name'] = null;
         }
+        else {
+            if (strlen($_POST['team_name']) > Team::MAX_TEAM_NAME_LENGTH) {
+                $this->core->addErrorMessage("Number of characters in team name must be less than " . Team::MAX_TEAM_NAME_LENGTH);
+                return new RedirectResponse($return_url);
+            }
+
+            if (preg_match('/[\x{1F600}-\x{1F64F}\x{1F300}-\x{1F5FF}\x{1F680}-\x{1F6FF}\x{1F700}-\x{1F77F}\x{1F780}-\x{1F7FF}\x{1F800}-\x{1F8FF}\x{1F900}-\x{1F9FF}\x{1FA00}-\x{1FA6F}\x{1FA70}-\x{1FAFF}\x{2600}-\x{26FF}\x{2700}-\x{27BF}\x{2300}-\x{23FF}]/u', $_POST['team_name'])) {
```

Comment:
Are these all of the possible special character values?  This should at least have a comment describing the purpose of this line.  Otherwise, is there a better way to go about doing this?  Instead of prohibiting characters, would it instead make more sense to create a regex which matches allowable characters?

Reply:
The above mentioned long regular expression checks for emojis in the team name.
It covers significant portion of the emoji characters.

I suppose this kind of blacklisting characters is well optimized than whitelisting allowable characters.